src/node_api.cc

```diff
@@ -23,12 +23,16 @@ void napi_clear_last_error(napi_env env);
 
 class napi_env__ {
  public:
-  explicit napi_env__(v8::Isolate* _isolate): isolate(_isolate), last_error() {}
+  explicit napi_env__(v8::Isolate* _isolate): isolate(_isolate),
+      has_instance_available(true), last_error() {}
   ~napi_env__() {
     last_exception.Reset();
+    has_instance.Reset();
   }
   v8::Isolate* isolate;
   v8::Persistent<v8::Value> last_exception;
+  v8::Persistent<v8::Value> has_instance;
```

Comment:
So, yes, it would work perfectly, and it's only our desire to be proactive that prompted us to implement this solution.